### PR TITLE
SNOW-1651271 Raised  for Index bitwise operators.

### PR DESCRIPTION
**Pull request by sfc-gh-azhan**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1651271

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: tests/integ/modin/index/test_index_methods.py [hunk 1]
```diff
@@ -42,6 +42,10 @@ def test_index_creation_from_lazy_index():
     assert_index_equal(i1, i2)
 
 
+def test():
+    pd.Index([1]) | 2
```
Comment (sfc-gh-nkrishna) [inline tests/integ/modin/index/test_index_methods.py:46]: What is this for? I don't see it used

File: src/snowflake/snowpark/modin/plugin/extensions/index.py [hunk 2]
```diff
@@ -332,6 +331,56 @@ def __le__(self, other: Any) -> Index:
     def __lt__(self, other: Any) -> Index:
         return self._binary_ops("lt", other)
 
+    def __or__(self, other: Any) -> Index:
+        ErrorMessage.not_implemented(
+            f"Index.{inspect.currentframe().f_code.co_name} is not yet implemented"
```
Comment (sfc-gh-mvashishtha) [inline src/snowflake/snowpark/modin/plugin/extensions/index.py:336]: why write the error message this way?

**Review by sfc-gh-mvashishtha — APPROVED** [review]
approved with optional comment

File: tests/integ/modin/index/test_index_methods.py (hunk 1 above)
Comment (sfc-gh-azhan) [reply tests/integ/modin/index/test_index_methods.py]: will remove.

File: src/snowflake/snowpark/modin/plugin/extensions/index.py (hunk 2 above)
Comment (sfc-gh-azhan) [reply src/snowflake/snowpark/modin/plugin/extensions/index.py:336]: easy to copy and paste :)

**Review by sfc-gh-nkrishna — APPROVED** [review]

File: src/snowflake/snowpark/modin/plugin/extensions/index.py [hunk 3]
```diff
@@ -248,9 +249,7 @@ def __getattr__(self, key: str) -> Any:
                 if hasattr(native_index, key):
                     # Any methods that not supported by the current Index.py but exist in a
                     # native pandas index object should raise a not implemented error for now.
-                    raise ErrorMessage.not_implemented(
-                        f"Index.{key} is not yet implemented"
-                    )
+                    ErrorMessage.not_implemented(f"Index.{key} is not yet implemented")
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/plugin/extensions/index.py:252]: Is `raise ErrorMessage.not_implemented()` equivalent to just `ErrorMessage.not_implemented()`, or was this a bug?

Comment (sfc-gh-nkrishna) [reply src/snowflake/snowpark/modin/plugin/extensions/index.py:252]: Just calling `ErrorMessage.not_implemented()` should handle the raising.

Comment (sfc-gh-helmeleegy) [reply src/snowflake/snowpark/modin/plugin/extensions/index.py:252]: But it used to have `raise` before it. So I'm suspecting this was a bug that was not captured by tests.

Comment (sfc-gh-azhan) [reply src/snowflake/snowpark/modin/plugin/extensions/index.py:252]: It was kind like a bug because `ErrorMessage.not_implemented` raise internally.

**Pull request merged**